### PR TITLE
NFT Studio: Collection preview images

### PR DESCRIPTION
**Pull request by onnovisser**
### Description

* Adds loading and error states
* Adds collection preview images
* Refetch collections and balance when creating collection

### Approvals

- [ ] Dev

### Screenshots

image

image

image

### Impact

NFT Studio, Fabric

### CODE REVIEW

File: nft-studio/src/components/CollectionCard.tsx [hunk 1]
```diff
@@ -1,32 +1,56 @@
...
         </Grid>
       </Shelf>
     </Card>
   )
 }
+
+const PreviewImage: React.FC<{ uri?: string }> = ({ uri }) => {
+  const { data } = useMetadata(uri)
+  return (
+    <Box bg="#eee" width="80px" height="80px" borderRadius={4} overflow="hidden">
```
Comment (alessioscalici) [inline nft-studio/src/components/CollectionCard.tsx:43]: If we hardcode colors, when we switch themes all the hardcoded colors will need to be updated.. Can we use theme colors here?

File: nft-studio/src/pages/Collections.tsx [hunk 2]
```diff
@@ -81,7 +105,7 @@ const TestTransaction: React.FC = () => {
         >
           Do some transaction
         </Button>
-        <div>Est. gas fee: {data}</div>
+        {/* <div>Est. gas fee: {data}</div> */}
```
Comment (alessioscalici) [inline nft-studio/src/pages/Collections.tsx:108]: Leftover comment

File: nft-studio/src/utils/useCollections.ts [hunk 3]
```diff
@@ -26,74 +26,75 @@ export type Collection = CollectionValue & {
...
+        const collectionValue = value.toJSON() as CollectionValue
+        const collection: Collection = {
+          id,
+          admin: encodeAddress(collectionValue.admin),
+          instances: collectionValue.instances,
+          metadataUri: metasObj[id]?.data,
+        }
+        return collection
+      })
+      console.log('collections', collections, metas, mapped)
```
Comment (alessioscalici) [inline nft-studio/src/utils/useCollections.ts:55]: All these console logs are ok now that we're deep in development, but I think they should be removed before the release, WDYT?

**Review by alessioscalici — APPROVED** [review]

File: nft-studio/src/components/CollectionCard.tsx (hunk 1 above)
Comment (onnovisser) [reply nft-studio/src/components/CollectionCard.tsx:43]: yea will make it a theme color

File: nft-studio/src/pages/Collections.tsx (hunk 2 above)
Comment (onnovisser) [reply nft-studio/src/pages/Collections.tsx:108]: will remove

File: nft-studio/src/utils/useCollections.ts (hunk 3 above)
Comment (onnovisser) [reply nft-studio/src/utils/useCollections.ts:55]: yea absolutely

**Pull request merged**